lib/navigation/filtering/running_means_filter.cpp

```diff
@@ -0,0 +1,25 @@
+#include "running_means_filter.hpp"
+
+namespace hyped::navigation {
+
+RunningMeansFilter::RunningMeansFilter(core::ILogger &logger, const core::ITimeSource &time)
+    : logger_(logger),
+      time_(time)
+{
+  acceleration_values_.resize(kNumDataPoints);
+  std::fill(acceleration_values_.begin(), acceleration_values_.end(), 0.0F);
```

Comment:
why is this done this way? seems weird to resize

Reply:
I mean it was this or write a loop? Is  there a better way of doing it?

Reply:
looking at this awake it makes sense now, you need a circular buffer and this makes sense to initialise it to a specific size.
You can do `acceleration_values_ = std::vector<core::Float>(length,initial_value);` but idk if that would be better form
Alternatively, if you do `acceleration_values_(length, initial_value)` in the initialiser list that also correctly initialises. I think this is the one I like most (but it is a bit more obtuse) but am open to arguments and am aware this is meaningless nitpicking

Reply:
I don't see anything directly wrong with current implementation? It just needs to be initialized to a certain size on class construction which is what it does?